app/celery/process_pinpoint_receipts_tasks.py

```diff
@@ -0,0 +1,145 @@
+from datetime import datetime
+
+from flask import current_app, json
+from notifications_utils.statsd_decorators import statsd
+from sqlalchemy.orm.exc import NoResultFound
+
+from app import notify_celery, statsd_client
+from app.config import QueueNames
+from app.dao import notifications_dao
+from app.models import (
+    NOTIFICATION_DELIVERED,
+    NOTIFICATION_PERMANENT_FAILURE,
+    NOTIFICATION_SENT,
+    NOTIFICATION_TECHNICAL_FAILURE,
+    NOTIFICATION_TEMPORARY_FAILURE,
+    PINPOINT_PROVIDER,
+)
+from app.notifications.callbacks import _check_and_queue_callback_task
+from celery.exceptions import Retry
+
+# Pinpoint receipts are of the form:
+#     {
+#         "eventType": "TEXT_DELIVERED",
+#         "eventVersion": "1.0",
+#         "eventTimestamp": 1712944268877,
+#         "isFinal": true,
+#         "originationPhoneNumber": "+13655550100",
+#         "destinationPhoneNumber": "+16135550123",
+#         "isoCountryCode": "CA",
+#         "mcc": "302",
+#         "mnc": "610",
+#         "carrierName": "Bell Cellular Inc. / Aliant Telecom",
+#         "messageId": "221bc70c-7ee6-4987-b1ba-9684ba25be20",
+#         "messageRequestTimestamp": 1712944267685,
+#         "messageEncoding": "GSM",
+#         "messageType": "TRANSACTIONAL",
+#         "messageStatus": "DELIVERED",
+#         "messageStatusDescription": "Message has been accepted by phone",
+#         "totalMessageParts": 1,
+#         "totalMessagePrice": 0.00581,
+#         "totalCarrierFee": 0.006
+#     }
+
+
+@notify_celery.task(bind=True, name="process-pinpoint-result", max_retries=5, default_retry_delay=300)
+@statsd(namespace="tasks")
+def process_pinpoint_results(self, response):
```

Comment:
Was this part of the resurrected code too? Not to be done right now, but as we want to move away from Celery processing the callbacks, I was wondering if that would not be a good opportunity to have pinpoint results processing all in lambda sooner with this work. What do you think?

Reply:
yes it is mostly the old code (and also pretty much a copy of the current sns/sms receipt processing).

I thought about moving it all to the new lambda, but figured that for a first step we just get this all in and working and then we could think about refactoring.

Admittedly, the pinpoint receipts would be a good first target if, for example, we restrict the shortcode to GCNotify initially.